Tella/Scenes/Resources/ViewModel/ResourcesViewModel.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  ResourcesViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/1/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+class ResourcesViewModel: ObservableObject {
+    @Published var downloadedResourcesVM : DownloadedResourcesVM
+    @Published var availableResourcesVM : AvailableResourcesVM
+    
+    init() {
+        self.downloadedResourcesVM = DownloadedResourcesVM()
```

Comment:
I think it's better to create one view model `ResourcesViewModel` that contains the code in the DownloadedResourcesVM and  AvailableResourcesVM because they are related and avoid passing in parameter later the downloadedResourcesVM to AvailableResourcesVM

Reply:
I separate the view models for clarity and modularity but makes sense to keep it in just one. Since I change these view models a lot then the others PRs I will add this change to [resources/downloaded](https://github.com/Horizontal-org/Tella-iOS/pull/124). Does that work for you?

Reply:
Yes sure

Reply:
I've been thinking about this and even though I agree that having one viewModel could simplify the data flow, we should also consider the principle of separation of concerns and modularity. Keeping DownloadedResourcesVM and AvailableResourcesVM separate allows us to encapsulate specific functionalities and responsibilities within distinct components, enhancing the readability and reusability of our code. It also makes it easier to manage, debug, and extend each aspect of resource management independently.

What we could do is to find a middle ground between the 2 alternatives. What I have in mind is for the resourceViewModel to act as the parent and orchestrate the interactions between both child view models, this way we could maintain modularity and also avoid passing downloadedResourcesVM to availableResourcesVM.

Let me know if this solution works for you, or if you still prefer the idea of creating just one view model. I'm open to talk about it!

Reply:
I my opinion in the current implentation there is an extra work (pass downloadedResourcesVM in parameter and fetchDownloadedResources) also I see that the two view models are related to each other and I don't see that 'DownloadedResourcesVM' is complex to avoid to include it together with `AvailableResourcesVM` 

After merging we will work on this view model to improve and simplify it .